liferay-headless/liferay-portal
# LPS-174435 Return fields from Rest API when getting nested fields

**Related issue**: https://issues.liferay.com/browse/LPS-174435

**How to reproduce:**

- Go to Control Panel -> Objects 
- Add a new custom object
- - Add an object field 
- - Publish the object
- Go to the User system object
- - Add a relationship with the recently created object 
- - - The relationship field will be created on the custom object side
- Submit some entries in the custom object 
- - Make sure to fill in the relationship field with some value, which should be a valid user ID
- Do the following request:

```
curl -X 'GET' \
  'http://localhost:8080/o/c/{customObjectName}?nestedFields=user' \
  -H 'accept: application/json' \
  -u 'test@liferay.com:test' 
```

**Expected Result:**

The `relationship` field value, for each object entry, should have the same format as described [here](https://liferay.slack.com/archives/C032VGYA2UD/p1676379359418929?thread_ts=1675359782.611929&cid=C032VGYA2UD), i.e., the format should be the same as the existing headless API format, to be consistent with current headless apis.

**OBS**.: I'm keeping the old behavior under FF.
